File: exercises/php/tests/tax/TaxCalculatorAfterFirstYearTest.php
<?php

namespace McrDigital\PhpFundamentals3\Tax;

use DateTime;
use PHPUnit\Framework\TestCase;

require_once "DummyTaxCalculator.php";

class TaxCalculatorAfterFirstYearTest extends TestCase
{
    private static DateTime $FIRST_OF_APRIL_2017;
    private TaxCalculator $taxCalculator;

    protected function setUp(): void
    {
        self::$FIRST_OF_APRIL_2017 = new DateTime("2017-04-01");
        $this->taxCalculator = new DummyTaxCalculator();
    }

    public function testSubsequentYearsTaxForPetrol(): void
    {
        $this->markTestSkipped('Ignored for now.');
        $vehicle = new Vehicle(206, FuelType::PETROL, self::$FIRST_OF_APRIL_2017, 20000);

        $this->assertEquals(140, $this->taxCalculator->calculateTax($vehicle));
    }

    public function testSubsequentYearsTaxForElectric(): void
    {
        $this->markTestSkipped('Ignored for now.');
        $vehicle = new Vehicle(206, FuelType::ELECTRIC, self::$FIRST_OF_APRIL_2017, 20000);

        $this->assertEquals(0, $this->taxCalculator->calculateTax($vehicle));
    }

    public function testSubsequentYearsTaxForAlternativeFuel(): void
    {
        $this->markTestSkipped('Ignored for now.');
        $vehicle = new Vehicle(206, FuelType::ALTERNATIVE_FUEL, self::$FIRST_OF_APRIL_2017, 20000);

        $this->assertEquals(130, $this->taxCalculator->calculateTax($vehicle));
    }
}

File: exercises/kotlin/src/test/kotlin/tax/TaxCalculatorAfterFirstYearTest.kt
package tax

import kotlin.test.Ignore
import org.junit.jupiter.api.Test
import java.time.LocalDate
import java.time.Month
import kotlin.test.assertEquals

@Ignore
class TaxCalculatorAfterFirstYearTest {

    private val taxCalculator: TaxCalculator = DummyTaxCalculator()
    private val FIRST_OF_APRIL_2017 = LocalDate.of(2017, Month.APRIL, 1)

    @Test
    fun subsequentYearsTaxForPetrol() {
        val vehicle = Vehicle(206, FuelType.PETROL, FIRST_OF_APRIL_2017, 20000)

        assertEquals(140, taxCalculator.calculateTax(vehicle))
    }

    @Test
    fun subsequentYearsTaxForElectric() {
        val vehicle = Vehicle(206, FuelType.ELECTRIC, FIRST_OF_APRIL_2017, 20000)
        assertEquals(0, taxCalculator.calculateTax(vehicle))
    }

    @Test
    fun subsequentYearsTaxForAlternativeFuel() {
        val vehicle = Vehicle(206, FuelType.ALTERNATIVE_FUEL, FIRST_OF_APRIL_2017, 20000)
        assertEquals(130, taxCalculator.calculateTax(vehicle))
    }
}